src/lsdb/loaders/hipscat/read_hipscat.py

```diff
@@ -66,6 +73,79 @@ def read_hipscat(
     return loader.load_catalog()
 
 
+def read_hipscat_subset(
+    path: str,
+    catalog_type: Type[Dataset] | None = None,
+    storage_options: dict | None = None,
+    search_filter: AbstractSearch | None = None,
+    order: int | None = None,
+    n_pixels: int | None = None,
+    **kwargs,
+) -> Dataset:
+    """Load a catalog subset from a HiPSCat formatted catalog.
+
+    To be used with a filter or (order, n_files), but not both.
+
+    Typical usage example, where we load a catalog from a cone search:
+        lsdb.read_hipscat_subset(
+            path="./my_catalog_dir",
+            catalog_type=lsdb.Catalog,
+            columns=["ra","dec"],
+            filter=lsdb.core.search.ConeSearch(ra, dec, radius_arcsec),
+        )
+
+    Typical usage example, where we load a catalog from a pre-defined number of files:
+         lsdb.read_hipscat_subset(
+            path="./my_catalog_dir",
+            catalog_type=lsdb.Catalog,
+            columns=["ra","dec"],
+            order=5,
+            n_pixels=10,
+        )
+
+    Args:
+        path (str): The path that locates the root of the HiPSCat catalog
+        catalog_type (Type[Dataset]): Default `None`. By default, the type of the catalog is loaded
+            from the catalog info and the corresponding object type is returned. Python's type hints
+            cannot allow a return type specified by a loaded value, so to use the correct return
+            type for type checking, the type of the catalog can be specified here. Use by specifying
+            the lsdb class for that catalog.
+        storage_options (dict): Dictionary that contains abstract filesystem credentials
+        search_filter (Type[AbstractSearch]): The filter method to be applied to the catalog.
+            By default, all files are considered.
+        n_pixels (int): The number of files to read, in case no filter was specified.
+            By default, all files are considered.
+        order (int): The order of the files to read, in case n_files was specified.
+            By default, the largest catalog order is used.
+        **kwargs: Arguments to pass to the read hipscat call
+
+    Returns:
+        Catalog object loaded from the given parameters
+    """
+    pixels_to_load = None
+    hc_structure = HCHealpixDataset.read_from_hipscat(path, storage_options)
+    pixels = hc_structure.get_healpix_pixels()
```

Comment:
I think the best way of structuring this to avoid loading the hipscat catalog twice would be to move the filtering logic into the `AbstractCatalogLoader` so that you can perform the filtering on the hc_structure after it's loaded there and use that as the hc_structure instead of making it here then making it again there.

Reply:
Thanks, good catch. This call is expensive and performing it twice was hindering the performance significantly!